lib/3scale_toolbox/proxy_logger.rb

```diff
@@ -0,0 +1,21 @@
+module ThreeScaleToolbox
+  class ProxyLogger
+    instance_methods.each do |meth|
+      # skipping undef of methods that "may cause serious problems"
+      undef_method(meth) if meth !~ /^(__|object_id)/
+    end
+
+    def initialize(subject)
+      @subject = subject
+    end
+
+    def method_missing(name, *args)
```

Comment:
When implementhing method_missing respond_to_missing should be also implemented.